lib/paper_trail/serializer.ex

```diff
@@ -144,4 +144,47 @@ defmodule PaperTrail.Serializer do
         "#{model_id}"
     end
   end
+
+  @spec serialize_model_changes(Ecto.Changeset.t()) :: map()
+  defp serialize_model_changes(%Ecto.Changeset{data: %schema{}} = changeset) do
+    field_values = serialize_model_field_changes(changeset)
+    embed_values = serialize_model_embed_changes(changeset)
+
+    field_values
+    |> Map.merge(embed_values)
+    |> schema.__struct__()
+  end
+
+  defp serialize_model_field_changes(%Ecto.Changeset{data: %schema{}, changes: changes}) do
+    change_keys = changes |> Map.keys() |> MapSet.new()
+
+    field_keys =
+      :fields
+      |> schema.__schema__()
+      |> MapSet.new()
+      |> MapSet.intersection(change_keys)
+      |> MapSet.to_list()
+
+    Map.take(changes, field_keys)
+  end
+
+  defp serialize_model_embed_changes(%Ecto.Changeset{data: %schema{}, changes: changes}) do
+    change_keys = changes |> Map.keys() |> MapSet.new()
+
+    embed_keys =
+      :embeds
+      |> schema.__schema__()
+      |> MapSet.new()
+      |> MapSet.intersection(change_keys)
+      |> MapSet.to_list()
+
+    changes
+    |> Map.take(embed_keys)
+    |> Map.new(fn {key, value} ->
+      case schema.__schema__(:embed, key) do
+        %Ecto.Embedded{cardinality: :one} -> {key, serialize_model_changes(value)}
+        %Ecto.Embedded{cardinality: :many} -> {key, Enum.map(value, &serialize_model_changes/1)}
```

Comment:
I'm currently trying to understand this line. Could you explain this further @maennchen , are these the only cases the code can ever hit? is there a _ -> _ case as well that allows certain keys to be ignored? or do they get ignored during embed_keys or maps with normal fields hit this `%Ecto.Embedded{cardinality: :one} -> {key, serialize_model_changes(value)}`  per each key?  Generally its a nice PR, but would have helped if we also published the changes in few PRs as I see few changes to are bit unrelated. Regardless Im currently reading up on everything, thanks for the very interesting work! We should document this functionality in the README.md as well before merging anything.

Reply:
There's only two cases: embeds_one and embeds_many. Therefore no catch-all is needed.

Reply:
There's four commits in this PR on purpose. I can open them in as many PRs as you like. (The order is kind of important though.)

Reply:
About documentation I don't think there's much to say. The library supports ectos features. The features are already documented at ecto.

What would you like me to mention in the README?

Reply:
@izelnakri I split the PR apart into #122, #123, #124

Reply:
Thank you! Merged the first two one, I'll also merge the third one. Could you make github clean the diff for the embed changes afterwards? I'll try to review in the next 10 days. Thanks again for all the work!

Reply:
@izelnakri Sure, I'll rebase it as well.

Reply:
@izelnakri Rebased as well.